src/posts/posts.service.ts

```diff
@@ -14,13 +14,46 @@ export class PostsService {
   /**
    * @description find all posts
    */
-  async getAllPosts(): Promise<Array<PostEntity>> {
+  async getAllPosts(
+    authorId?: string,
+    hashtags?: string[] | null,
+  ): Promise<Array<PostEntity>> {
     // TODO: implementation pagination (size + limit)
-    // TODO: implement filter by author
     // TODO: implement filter by hashtag
-    return this.postsRepository.find({
-      take: 100,
-      order: { createdAt: 'DESC' },
+    const queryBuilder = this.postsRepository.createQueryBuilder('posts');
+
+    if (authorId) {
+      queryBuilder
+        .leftJoinAndSelect('posts.author', 'author')
+        .where(`posts.author = :authorId`, { authorId })
+        .addSelect('posts.created_at')
+        .orderBy('posts.created_at', 'DESC');
```

Comment:
can we keep the orderBy for createdAt outside this if condition ? 

Let's only add 

```
queryBuilder
        .leftJoinAndSelect('posts.author', 'author')
        .where(`posts.author = :authorId`, { authorId })
        .addSelect('posts.created_at')
```

In this if block.

Reply:
Sure thing! Addresses that.